### ISSUE TITLE
Implement Note linking between notes in the note editor.

### ISSUE THREAD

**Issue by timoc** [d1vanov/quentier#186]
Implement Note linking as described here:
https://evernote.com/blog/use-note-links-to-get-around-faster/

With drag and drop from the note selector, as an option.

**Comment by d1vanov** [comment]
A good suggestion, thanks. In fact, currently a small part of it is implemented: clicking on any pre-existing note link should cause the corresponding note to be open as a new note editor tab. But insertion of links and its convenient variant via drag-n-drop would also be good to have.

**Comment by robert7** [comment]
I just looked at it, at it actually works/is implemented already - just the drag & drop is missing. But I would see the drag&drop as kind of "nice to have".

Another related thing which would be nice to have, is to have some visual hint for "intra note links" to visually distinguish them from external links. Those used to be "green" in Evernote - so intra links were green and external were blue by default. But I think the "f-cked up" in the last versions and in Web client now all links are blue.

@timoc 
here how it works **now**:
![screen_20180721_22](https://user-images.githubusercontent.com/1135218/43038897-72c28966-8d22-11e8-8c7d-03be6d4793e7.png)

![screen_20180721_25](https://user-images.githubusercontent.com/1135218/43038899-770b4b84-8d22-11e8-9726-fcb5a90799e3.png)

![screen_20180721_26](https://user-images.githubusercontent.com/1135218/43038900-7b53f4fc-8d22-11e8-9f45-d216afc954d1.png)

**Comment by robert7** [comment]
@d1vanov 
I would suggest to rename the context menu option. Because it may be confusing. It actually not "in-app" (which would suggest it only works in Quentier). It will work with all Evernote clienst.
In Evernote its simply called "Copy notelink" => so I guess users will expect this.

![screen_20180721_27](https://user-images.githubusercontent.com/1135218/43038932-1223a22e-8d23-11e8-9940-180749d4d16d.png)

**Comment by d1vanov** [comment]
I used that name - "in-app note link" - because it is actually Evernote term as explained [here](https://dev.evernote.com/doc/articles/note_links.php). Basically, there are other kinds of note links besides these ones, so these are called "in-app".

**Comment by robert7** [comment]
Yes but  "in-app note link" is kind of technical term. Evernote doesn't use it in UI in its applications (never seen that). So thats what I mean, that users of existing Evernote apps,  can be confused by this.

**Comment by timoc** [comment]
Thanks. That i had to post to github before i could make a note link (it was difficult to find). I did find it afterwards myself, but i would have expected it to be available in more menus than just the context menu for the note.
Drag and drop would also be great IMHO. If you wanted to be smarter about it, you could pop up a menu to confirm, or link to headlines or something.
